File: javascript/observer/WeatherRama/observer/displays/CurrentConditionDisplay.js
import {Observer} from "../Observer.js";
import {WeatherStation} from "../../subject/WeatherStation.js";

export class CurrentConditionDisplay extends Observer {

    // private variables
    #temperature = 0
    #humidity = 0
    #pressure = 0

    #weatherStation = new WeatherStation()

    constructor(weatherStation) {
        super();
        this.#weatherStation = weatherStation
        this.#weatherStation.registerObserver(this)
    }

    update(temperature, humidity, pressure) {
        super.update(temperature, humidity, pressure);
        this.#temperature = temperature
        this.#humidity = humidity
        this.#pressure = pressure
        this.display()
    }

    display() {
        console.log("CurrentCondition:\n"
            + this.#temperature + "F degree\n"
            + this.#humidity + "% humidity\n"
            + this.#pressure + " pressure\n"
        )
    }

}

File: ruby/observer/WeatherRama/observer/displays/current_condition_display.rb
require './observer/display'
require './observer/observer'
require './subject/weather_station'

class CurrentConditionDisplay
  include Observer
  include Display

  def initialize(weather_station)
    @weather_station = WeatherStation.new
    @weather_station = weather_station
    @weather_station.register_observer(self)
    @temperature = 0
    @humidity = 0
    @pressure = 0
  end

  def update(temperature, humidity, pressure)
    super
    @temperature = temperature
    @humidity = humidity
    @pressure = pressure
    display
  end

  def display
    super
    puts("Current Conditions:")
    puts(@temperature.to_s + "F degree")
    puts(@humidity.to_s + "% humidity")
    puts(@pressure.to_s + " pressure")
  end

end